### ISSUE TITLE
proposal: `multiple_show_or_hide`

### ISSUE THREAD

**Issue by eernstg** [dart-lang/linter#4052]
## multiple_show_or_hide

A lint against using constructs like `show ... show ...` or `show ... hide ...` in imports or exports.

## Description

Use at most one `show` or `hide` clause in an import or export.

## Details

Multiple `show` clauses, multiple `hide` clauses, or both, in an import or export directive have a convoluted and counter-intuitive semantics. A single `show` clause or a single `hide` clause can always express the same thing as any given multiple-clause directive, and the former is more concise and more readable.

## Kind

This lint enforces a style: Import and export directives should use a single `show` or `hide` clause, rather than expressing the same thing using multiple such clauses.

This lint guards against errors: The multiple-clause form has a convoluted and counter-intuitive semantics, and a reader of the code is likely to misinterpret how it works.

## Bad Examples

```dart
import 'dart:core' show print show int hide Future;

void main() {
  int i = 1; // Error, unknown identifier `int`.
  print(i); // Error, unknown identifier `print`.
}
```

This import directive does not import anything: The first show clause imports `print`, the second show clause turns that into the empty set, and the hide clause has no effect (because it makes no difference that we remove `Future` from the empty set).

## Good Examples 

```dart
import 'dart:core' show print, int;

void main() {
  int i = 1; // OK.
  print(i); // OK.
}
```

## Discussion

To the best of my knowledge, this lint does not conflict with any other lints or other diagnostics, and does not overlap or modify any such rules.

Two relevant issues are: https://github.com/dart-lang/sdk/issues/32878, https://github.com/dart-lang/language/issues/2824.

### Discussion checklist

- [x] List any existing rules this proposal modifies, complements, overlaps or conflicts with.
- [x] List any relevant issues (reported here, the [SDK Tracker], or elsewhere).
- [ ] If there's any prior art (e.g., in other linters), please add references here.
- [ ] If this proposal corresponds to [Effective Dart] or [Flutter Style Guide] advice, please call it out. (If there isn't any corresponding advice, should there be?)
- [ ] If this proposal is motivated by real-world examples, please provide as many details as you can.  Demonstrating potential impact is _especially valuable_.

[Effective Dart]: https://dart.dev/guides/language/effective-dart
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[SDK Tracker]: https://github.com/dart-lang/sdk/issues

**Comment by bwilkerson** [comment]
See https://github.com/dart-lang/language/issues/2824#issuecomment-1421164219.

**Comment by lrhn** [comment]
I don't think this should be a lint. If we warn, it should be an always-on analyzer warning about unnecessary code.
You never need more than one `hide` or `show`, it's impossible to have a false positive that can't be fixed by combining into one modifier.
